### DISCUSSION TITLE
exchangelib.errors.TransportError: Failed to get auth type from service

### DISCUSSION THREAD

**Discussion by bjs94** [ecederstrand/exchangelib#1021] [Q&A]
I'm trying to connect to outlook and read email. I'm using an automated company email to fetch the inbox. If I include auth_type it gives credential errors even though the credentials are correct. This account doesn't prompt any authentication like 2-factor or anything although my company has it, so I removed authentication auth_type under config for this. This is my code, 

```python
from exchangelib import DELEGATE, Account, Credentials, Configuration
credentials = Credentials(
    username='john@example.com',
    password='topsecret'
)
ews_url = 'https://outlook.office365.com/owa/company.org/EWS/Exchange.asmx'
primary_smtp_address = 'john@example.com'

config = Configuration(service_endpoint=ews_url, credentials=credentials)

account = Account(
    primary_smtp_address=primary_smtp_address,
    config=config,
    autodiscover=False,
    access_type=DELEGATE
)

for item in account.inbox.all().order_by('-datetime_received')[:10]:
    print(item.subject, item.sender, item.datetime_received)
```

This is the error I get,

```
Failed to create cached protocol with key ('https://outlook.office365.com/owa/company.org/EWS/Exchange.asmx', Credentials(''john@example.com', 'topsecret')): Failed to get auth type from service
Traceback (most recent call last):
  File "C:/mycode.py", line 28, in <module>
    account = Account(
  File "C:\Python\Python38\lib\site-packages\exchangelib\account.py", line 126, in __init__
    self.protocol = Protocol(config=config)
  File "C:\Python\Python38\lib\site-packages\exchangelib\protocol.py", line 411, in __call__
    raise e
  File "C:Python\Python38\lib\site-packages\exchangelib\protocol.py", line 406, in __call__
    protocol = super().__call__(*args, **kwargs)
  File "C:Python\Python38\lib\site-packages\exchangelib\protocol.py", line 452, in __init__
    self.config.auth_type = self.get_auth_type()
  File "C:\Python\Python38\lib\site-packages\exchangelib\protocol.py", line 457, in get_auth_type
    auth_type, api_version_hint = get_service_authtype(
  File "C:\Python\Python38\lib\site-packages\exchangelib\transport.py", line 165, in get_service_authtype
    raise TransportError('Failed to get auth type from service')
exchangelib.errors.TransportError: Failed to get auth type from service
```

Please help me, I've been trying for days and couldn't figure this out.

**Comment by ecederstrand** [comment]
Your EWS URL value looks wrong. exchangelib cannot use OWA to connect. I suggest you use `server='outlook.office365.com'` instead.

**Reply by bjs94** [reply]
I changed it to what you said. It still gives me the same error as above.

**Reply by ecederstrand** [reply]
If you changed the server, it can't have given you the exact same error message. At least the URL will have changed.

It's possible that your credentials are not correct, or that your credentials are not allowed to use EWS in your company's O365 configuration.

Otherwise, enabling [debug logging](https://ecederstrand.github.io/exchangelib/#troubleshooting) may give you more information about why the connection does not succeed.

**Reply by bjs94** [reply]
Thank you. So tried with autodiscover and it's working on my coworker's machine. I had to install exchangelib 3.2.0 since that's the compatible version for our server. However I get this error on mine.

C:\Python\Python38\lib\site-packages\exchangelib\ewsdatetime.py:251: PytzUsageWarning: The zone attribute is specific to pytz's interface; please migrate to a new time zone provider. For more details on how to do so, see https://pytz-deprecation-shim.readthedocs.io/en/latest/migration.html
  self_cls.ms_id = cls.PYTZ_TO_MS_MAP[tz.zone][0]
Traceback (most recent call last):
  File "C:/mycode.py", line 9, in <module>
    account = Account(
  File "C:\Python\Python38\lib\site-packages\exchangelib\account.py", line 92, in __init__
    self.default_timezone = default_timezone or EWSTimeZone.localzone()
  File "C:\Python\Python38\lib\site-packages\exchangelib\ewsdatetime.py", line 271, in localzone
    return cls.from_pytz(tz)
  File "C:\Python\Python38\lib\site-packages\exchangelib\ewsdatetime.py", line 260, in from_pytz
    self = self_cls()
TypeError: __init__() missing 2 required positional arguments: 'zone' and 'key'

**Reply by bjs94** [reply]
The issue is fixed, thank you for this lib!

**Reply by kurisutofu** [reply]
> The issue is fixed, thank you for this lib!

How did you fix the issue? I'm seeing the same in my code.

**Reply by kurisutofu** [reply]
> > The issue is fixed, thank you for this lib!
> 
> How did you fix the issue? I'm seeing the same in my code.

Nevermind ... the version of python was not correct. After moving to 3.8, the error disappeared.

**Reply by ernekyan** [reply]
Hi, i am on Python 3.9 and the issue is still here =)

**Reply by ecederstrand** [reply]
@ernekyan Which issue are you referring to?

This question is more than two years old. I think it's better to open a new question with the detauls of your issue.

**Reply by ernekyan** [reply]
@ecederstrand thank for response, i have Transport error , exchangelib 4.7.2, after 2-3 running 
```
File "C:\Program Files\Python39\lib\site-packages\exchangelib\transport.py", line 173, in get_service_authtype 
    raise TransportError(str(e)) from e 
exchangelib.errors.TransportError: ('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None))
```
copy my code in another file and it work 2-3 times and again
i think something wrong with session, but i already use account.protocol.close() and have no idea 
should i open a new question?

**Reply by ecederstrand** [reply]
@ernekyan That's your server or network connection randomly aborting connections. Not related to any of the above issues. If you want to handle those errors gracefully, set up a retry policy as described in the documentation.
